buildtest/buildsystem/base.py

```diff
@@ -455,31 +436,35 @@ def run_tests(self, testfile):
             self.metadata.get("rundir"), "%s.out" % self.build_id
         )
 
-        # Run the test file, print output to file
-        with open(run_output_file, "w") as fd:
+        # Keep an error file
+        run_error_file = os.path.join(
```

Comment:
For this, it would be more elegant to create a run_output file:

```python
run_output_file = os.path.join(	
            self.metadata.get("rundir"), self.build_id
        )
```
And then using the basename for each of .out and .err:

```python
with open(run_output_file + ".err")
   # write error

with open(run_output_file + ".out")
   # write output
```
I don't see from these changes where run_output_file and run_error_file are coming from, but that could be adopted to the above.

Reply:
This makes sense i can refactor this stay tuned, it came to my mind also but didn't do it for first implementation.

Reply:
Sounds good!